packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -93,10 +94,25 @@ export function canBindAttribute( blockName, attributeName ) {
 export const withBlockBindingSupport = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
 		const registry = useRegistry();
-		const sources = useSelect( ( select ) =>
-			unlock( select( blocksStore ) ).getAllBlockBindingsSources()
-		);
 		const { name, clientId, context } = props;
+		const { sources, hasParentPattern } = useSelect(
+			( select ) => {
+				const { getAllBlockBindingsSources } = unlock(
+					select( blocksStore )
+				);
+				const { getBlockParentsByBlockName } = unlock(
+					select( blockEditorStore )
+				);
+
+				return {
+					sources: getAllBlockBindingsSources(),
+					hasParentPattern:
+						getBlockParentsByBlockName( clientId, 'core/block' )
```

Comment:
This adds a block editor store subscription for all bindable block, no? Any early returns we can set here to avoid it?

Reply:
I also noticed this lookup for a parent pattern block has been added in a few places, another example here and there are more for the UI parts of pattern overrides - https://github.com/WordPress/gutenberg/blob/7bca2fadddde7b2b2f62823b8a4b81378f117412/packages/editor/src/bindings/pattern-overrides.js#L16-L20

We could look into using block context as a way of checking for a parent pattern, though it'd mean every (supported) block needs to subscribe to that context.

There isn't an obvious context attribute that the pattern block could provide. There's only `ref` (which may be removed in the future when the block uses a `slug`), and `content` (which can be `undefined` so might not reliable, unless it's changed so a default value is provided).

Reply:
I made a PR based on this that shows how that would work - https://github.com/WordPress/gutenberg/pull/62861.

Reply:
I answered you there, but I think using context makes sense. Thanks for working on that! 🙂